holoviews/plotting/bokeh/element.py

```diff
@@ -1189,6 +1192,11 @@ def _get_color_data(self, element, ranges, style, name='color', factors=None, co
 
         mapper = self._get_colormapper(cdim, element, ranges, style,
                                        factors, colors)
+        if not factors and isinstance(mapper, CategoricalColorMapper):
+            field += '_str'
```

Comment:
Looks a little hackish?

Reply:
Does it? In what way?

Reply:
To be clearer, what's your suggestion for not overriding the original column and making it clear the column has been "categorized" by casting values to strings?